iml-services/iml-device/src/lib.rs

```diff
@@ -402,14 +418,24 @@ pub fn find_targets<'a>(
         .collect();
 
     xs.into_iter()
-        .map(|(fqdn, ids, mntpnt, fs_uuid, target)| Target {
-            state: "mounted".into(),
-            active_host_id: Some(*fqdn),
-            host_ids: ids,
-            filesystems: target_to_fs_map.get(target).cloned().unwrap_or_default(),
-            name: target.into(),
-            uuid: fs_uuid.into(),
-            mount_path: Some(mntpnt.0.to_string_lossy().to_string()),
+        .map(|(fqdn, ids, mntpnt, fs_uuid, target)| {
+            let filesystems: Vec<String> = target_to_fs_map
+                .get(target)
+                .cloned()
+                .unwrap_or_default()
+                .into_iter()
+                .filter(|x| snapshots.iter().find(|s| s.contains(x)).is_none())
```

Comment:
Why does this need to be checked here? Why can't we just filter the snapshots directly in this iterator?

Reply:
In this case I have the following:
```
target_to_fs_map.get("MGS") --> vec!["35efda93", "zfsmo"]
snapshots --> vec!["35efda93-MDT0000"]
```
I needed a way to filter out snapshots from this list. So I checked each item against the snapshots. Do you suggest a different approach?

Reply:
Probably before we get to this point. I.E. the target_to_fs_map should not have any snapshots in it's fs list.

Reply:
Ok, I'll see if I can filter them out before we call this function.